tests/integration/Core/Repository/Values/User/Limitation/ChangeOwnerLimitationTest.php

```diff
@@ -0,0 +1,192 @@
+ */
+namespace Ibexa\Tests\Integration\Core\Repository\Values\User\Limitation;
+
+use ErrorException;
+use Ibexa\Contracts\Core\Repository\ContentService;
+use Ibexa\Contracts\Core\Repository\Exceptions\UnauthorizedException;
+use Ibexa\Contracts\Core\Repository\PermissionResolver;
+use Ibexa\Contracts\Core\Repository\RoleService;
+use Ibexa\Contracts\Core\Repository\Values\Content\Content;
+use Ibexa\Contracts\Core\Repository\Values\User\Limitation\ChangeOwnerLimitation;
+use Ibexa\Contracts\Core\Repository\Values\User\PolicyDraft;
+use Ibexa\Contracts\Core\Repository\Values\User\Role;
+use Ibexa\Contracts\Core\Repository\Values\User\RoleDraft;
+use Ibexa\Contracts\Core\Repository\Values\User\User;
+
+/**
+ * @covers \Ibexa\Contracts\Core\Repository\Values\User\Limitation\ChangeOwnerLimitation
+ * @group integration
+ * @group limitation
+ */
+final class ChangeOwnerLimitationTest extends BaseLimitationTest
+{
+    private PermissionResolver $permissionResolver;
+
+    private RoleService $roleService;
+
+    private ContentService $contentService;
+
+    protected function setUp(): void
+    {
+        $repository = $this->getRepository();
+
+        $this->permissionResolver = $repository->getPermissionResolver();
+        $this->roleService = $repository->getRoleService();
+        $this->contentService = $repository->getContentService();
+    }
+
+    /**
+     * @return array<array{
+     *     0: \Ibexa\Contracts\Core\Repository\Values\User\User,
+     *     1: int,
+     *     2: int[]
+     * }>
+     */
+    public function allowedProvider(): array
+    {
+        $otherUserId1 = 123;
+        $otherUserId2 = 456;
+
+        return [
+            [null, []],
+            [null, [-1]],
+            [null, [$otherUserId1, $otherUserId2, -1]],
+            [$otherUserId1, [$otherUserId1, $otherUserId2, -1]],
+            [$otherUserId1, []],
+        ];
+    }
+
+    /**
+     * @return array<array{
+     *     0: \Ibexa\Contracts\Core\Repository\Values\User\User,
+     *     1: int,
+     *     2: int[]
+     * }>
+     */
+    public function deniedProvider(): array
+    {
+        $otherUserId1 = 123;
+        $otherUserId2 = 456;
+        $otherUserId3 = 789;
+
+        return [
+            [$otherUserId1, [-1]],
+            [$otherUserId1, [$otherUserId2, $otherUserId3]],
+        ];
+    }
+
+    /**
+     * @dataProvider allowedProvider
+     *
+     * @param int[] $limitationValues
+     */
+    public function testChangeOwnerLimitationAllowed(?int $ownerId, array $limitationValues): void
+    {
+        /* BEGIN: Use Case */
+        $currentUser = $this->createUserVersion1('current_user', null, null, 42);
+        $ownerId = $ownerId ?? $currentUser->id;
+
+        $draft = $this->createTestCaseDraft($currentUser, $ownerId, $limitationValues);
+        /* END: Use Case */
+
+        self::assertInstanceOf(Content::class, $draft);
```

Comment:
This assertion is dead because if `createTestCaseDraft` won't return an instance of `Content`, a fatal error will occur earlier. What you're looking for is:
```suggestion
        $this->expectNotToPerformAssertions();
```